src/main/scala/esmeta/analyzer/TypeAnalyzer.scala

```diff
@@ -161,7 +175,10 @@ class TypeAnalyzer(
       if (config.arity && (len < from || to < len))
         addMismatch(ArityMismatch(cp, len))
       // fill optional args after arity checked
-      val argsWithOptional = args ++ List.fill(to - len)(AbsValue.absentTop)
+      val viewargs = calleeNp.view.tys.map { AbsValue(_) }
+      val argsWithOptional = (if (TY_SENS) viewargs else args) ++ List.fill(
```

Comment:
This currently has problem of disabling all ParamTypeMismatch as when actual args are more coarse than expected viewargs become expected.

Reply:
This comment is no longer after force push.